packages/core/lib/span-factory.ts

```diff
@@ -51,6 +51,10 @@ export class SpanFactory {
     this.openSpans = new WeakSet<SpanInternal>()
   }
 
+  get firstSpanContext () {
+    return this.spanContextStorage.first
+  }
```

Comment:
I think it'd be better to expose `spanContextStorage` to plugins in the same way the span factory is

i.e. change `options.plugins`:

```diff
-plugins: (spanFactory: SpanFactory) => Array<Plugin<C>>
+plugins: (spanFactory: SpanFactory, spanContextStorage: SpanContextStorage) => Array<Plugin<C>>
```

Alternatively make `options.spanContextStorage` not optional so `browser.ts` can pass it in directly

Reply:
oh, that's a much nicer idea! I can definitely see the use in it being available to other plugins - going with option 1